### PR TITLE
{devel}[gompi/2019b] Boost v1.71.0

### PR DESCRIPTION
**Pull request by smoors**
(created using `eb --new-pr`)
needs https://github.com/easybuilders/easybuild-easyblocks/pull/1814

### CODE REVIEW

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node318.hydra.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/2a3ab2c5a4c32c1e76380e82ba94c229 for a full test report.

File: easybuild/easyconfigs/b/Boost/Boost-1.71.0-gompi-2019b.eb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name = 'Boost'
+version = '1.71.0'
+
+homepage = 'http://www.boost.org/'
```
Comment (Micket) [inline easybuild/easyconfigs/b/Boost/Boost-1.71.0-gompi-2019b.eb]: https

File: easybuild/easyconfigs/b/Boost/Boost-1.71.0-gompi-2019b.eb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+sources = ['%%(namelower)s_%s.tar.gz' % '_'.join(version.split('.'))]
+checksums = ['96b34f7468f26a141f6020efb813f1a2f3dfb9797ecf76a7d7cbd843cc95f5bd']
+
+dependencies = [
+    ('bzip2', '1.0.8'),
+    ('zlib', '1.2.11'),
+    ('XZ', '5.2.4'),
+]
+
+configopts = '--without-libraries=python'
```
Comment (Micket) [inline easybuild/easyconfigs/b/Boost/Boost-1.71.0-gompi-2019b.eb]: I know this has come up before; do we still want to go with this approach? I've heard this split causes problems for some software that expects a single BOOST_PATH. With python @ core, it wouldn't be to bad to include it here (granted, with the python version suffix).

I don't have a strong opinion either way, but, worth asking.

Comment (boegel) [reply easybuild/easyconfigs/b/Boost/Boost-1.71.0-gompi-2019b.eb]: Which Python version would you pick then? Or do we go back to having two `Boost` easyconfigs, one for Py2, one for Py3 (which basically means dragging in Python as a dep everywhere where `Boost` is required).

Related (which complicates things): #8810

Comment (smoors) [reply easybuild/easyconfigs/b/Boost/Boost-1.71.0-gompi-2019b.eb]: @Micket would the suggestion made by @boegel in #8810 (symlinking everything from Boost into Boost.Python) solve the problems with BOOST_PATH that you mentioned?

Comment (Micket) [reply easybuild/easyconfigs/b/Boost/Boost-1.71.0-gompi-2019b.eb]: @smoors alright that sounds promising.

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
vera-c1 - Linux centos linux 7.6.1810, Intel Xeon Processor (Skylake), Python 2.7.5
See https://gist.github.com/72545d8ccf83827d151f2eae9cbbf019 for a full test report.

**Comment by Micket** [comment]
Download isn't working; the 1.71 directory on sourceforge is empty? https://sourceforge.net/projects/boost/files/boost/1.71.0/

**Comment by smoors** [comment]
@Micket indeed, that's odd.
anyway, switching to their "official" source url (checksum remains the same).

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
vera-c1 - Linux centos linux 7.6.1810, Intel Xeon Processor (Skylake), Python 2.7.5
See https://gist.github.com/31c658e2f33660054428c40dfbf84630 for a full test report.

**Comment by Micket** [comment]
`/bin/bash: ./bjam: No such file or directory`

**Comment by smoors** [comment]
@Micket yes, please check my first comment on this PR ;)

**Comment by zao** [comment]
Heh, `b2` has been the recommended name of the Boost.Build binary for ages. Nice to see that they finally got rid of the `bjam` one :D

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
vera-c1 - Linux centos linux 7.6.1810, Intel Xeon Processor (Skylake), Python 2.7.5
See https://gist.github.com/acb0a41c10e76e9c2c88cd5fede99d74 for a full test report.

**Comment by Micket** [comment]
Going in, thanks @smoors!

**Pull request merged**